### ISSUE TITLE
Pytorch nn 과 nn.functional

### ISSUE THREAD

**Issue by onjsdnjs** [djkim1991/pytorchStudy#3]
### 필요 패키지
```
import torch.nn as nn
import torch.nn.functional as F
```

### 문서
- nn
https://pytorch.org/docs/1.0.1/nn.html
- nn.functional
https://pytorch.org/docs/1.0.1/nn.html#torch-nn-functional
---
### nn 과 nn.functional
제공하는 기능들은 비슷한데 사용하는 방법에 차이가 있습니다.

**nn 이 제공하는 기능들**
- Parameters
- Containers
- Conv
- Pooling
- Padding
- Non-linear Activation
- Normalization
- Recurrent
- Linear
- Dropout
- Sparse
- Distance
- Loss
- Vision
- Data Parallel
- Utilities
- ...

### nn.functional 이 제공하는 기능들
- Conv
- Pooling
- Non-linear Activation
- Normalization
- Linear function(=fully connected layer)
- Dropout
- Loss
- Vision
- ...
---
### conv2d의 사용
nn 과 nn.functional의 차이를 conv2d를 이용해서 알아봅시다.
```
nn.conv2d
nn.functionnal.conv2d
```
![](https://icecreamlabs.com/wp-content/uploads/2018/08/33-con.gif)

---

### torch.nn.Conv2d VS torch.nn.functional.Conv2d
입력 이미지에 2d Convolution을 적용시키는 메소드 입니다.
```
torch.nn.Conv2d(in_channels, out_channels, kernel_size, stride=1, padding=0, dilation=1, groups=1, bias=True)
```
```
torch.nn.functional.conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1)
```
### `nn.Conv2d`
> weight를 직접 선언해주지 않음
- `in_channels` : input image 채널의 수
- `out_channels` : convolution 결과 채널의 수
- `kernel_size` : convolution 커널의 크기(필터 크기)

### `torch.nn.functional.conv2d`
> weight를 직접 선언 ( 외부에서 만든 filter를 사용 )
- `input` : input Tensor ( minibatch×in_channels×iH×iW )
- `weight` : 필터의 shape ( out_channels × groups in_channels × kH × kW )
---
참고: 김군이(https://www.youtube.com/watch?v=VKhFeh92eps)
